code/modules/projectiles/projectile/bullets.dm

```diff
@@ -343,4 +339,54 @@
 	pellets = 6
 	range_step = 1
 	spread_step = 10
-	silenced = TRUE
\ No newline at end of file
+	silenced = TRUE
+
+/* toy projectiles */
+/obj/item/projectile/bullet/cap
+	name = "cap"
+	desc = "SNAP!"
+	damage = 0 // It's a damn toy.
+	embed_chance = 0
+	nodamage = TRUE
+	sharp = FALSE
+	damage_type = HALLOSS
+	impact_effect_type = null
+	fire_sound = 'sound/effects/snap.ogg'
+	combustion = FALSE
+
+/obj/item/projectile/bullet/cap/process()
+	loc = null
+	qdel(src)
+
+/obj/item/projectile/bullet/foam_dart
+	name = "foam dart"
+	desc = "I hope you're wearing eye protection."
+	damage = 0 // It's a damn toy.
+	embed_chance = 0
+	nodamage = TRUE
+	sharp = FALSE
+	damage_type = HALLOSS
+	impact_effect_type = null
+	fire_sound = 'sound/items/syringeproj.ogg'
+	combustion = FALSE
+	icon = 'icons/obj/gun_toy.dmi'
+	icon_state = "foamdart_proj"
+	range = 15
+
+/obj/item/projectile/bullet/foam_dart/on_impact(var/atom/A)
+	. = ..()
+	var/turf/T = get_turf(loc)
+	if(istype(T))
+		new /obj/item/ammo_casing/afoam_dart(get_turf(loc))
+
+/obj/item/projectile/bullet/foam_dart/on_range(var/atom/A)
+	. = ..()
+	var/turf/T = get_turf(loc)
+	if(istype(T))
+		new /obj/item/ammo_casing/afoam_dart(get_turf(loc))
+
+/obj/item/projectile/bullet/foam_dart/riot
+	name = "riot foam dart"
+	desc = "Whose smart idea was it to use toys as crowd control? Ages 18 and up."
+	agony = 50

```

Comment:
Actual, powerful stun weaponry just being easily available as vendor contraband doesn't seem like too good of an idea. I know vendor isn't mapped anywhere, but it seems like idea is that it will be eventually.

Reply:
Removed